### DISCUSSION TITLE
[Unknown Mod][Combat] Option to control the Catapult during combat

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2774] [Ideas]
Add an option to control the Catapult during combat:

"Battle: allows to control Catapult during combat"

When the Catapult has to shoot, this is visually indicated to the player.
Each Turret (the 2 turrets), the Ballista, and each part of the Wall (the 4 portions of the wall) can be targeted independently.
The player selects the target by moving the mouse cursor over the target.
The selected target is visually indicated to the player.
The player left-clicks on the target to activate the shot.

A test is then performed to determine whether the chosen target will be hit.

The percentages of chance to hit are as follows:
- Ballista: 25%
- Door (and drawbridge): 33%
- Turrets (2 turrets): 33%
- Walls (4 walls): 50%

The standard damage (or no damage) is then applied as usual.
A successful shot test does not necessarily mean that there will be damage for sure.

The percentages of success to hit the target should be accessible in a file external to the game so that they can be easily modified.

The hero does not need to have the Ballistics skill to be able to control the Catapult.
The player controls the Catapult in all cases when the option is activated.

When the Catapult has to shoot, the hero cannot cast a spell.
The hero can cast a spell only when he controls the troop that has the hand whether it is an allied troop or an opposing troop (for example under the influence of the Hypnotize spell).

The player can skip the turn of the Catapult (do not shoot).

In all cases, it is not possible to cast a spell on the Catapult.

**Note: all of this could eventually change.**

**Comment by Branikolog** [comment]
Well, control of catapult in Heroes of Might and Magic 3 was really a good improvement. And I would also like to see something similar to it in our "beyond 1.0" versions. Even more simple mechanics would make more fun with this skill. 
In my personal point, it would be nice if player can at least choose one of lower or upper parts of castle should be attacked by catapult. In some cases it can bring certain advantages.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Well, control of catapult in Heroes of Might and Magic 3 was really a good improvement. And I would also like to see something similar to it in our "beyond 1.0" versions. Even more simple mechanics would make more fun with this skill.
> In my personal point, it would be nice if player can at least choose one of lower or upper parts of castle should be attacked by catapult. In some cases it can bring certain advantages.

This is the goal.
In the described option, each Turret (the 2 turrets) and each part of the Wall (the 4 portions of the wall) can be targeted independently.
I specified this in the original post as it didn't seem clear.

**Reply by Branikolog** [reply]
So you're proposing to make the same mechanics to HoMM3, right?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> So you're proposing to make the same mechanics to HoMM3, right?

Similar but different and better than in Heroes 3, because the Ballistics skill is not required and is independent in its operation.

**Reply by Branikolog** [reply]
All later HoMM series don't allow to control Ctapult without a skill. Should we better make the expected machanics then?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> All later HoMM series don't allow to control Ctapult without a skill. Should we better make the expected machanics then?

It will work this way in my mod.

You can maybe propose an option (the two options would be exclusive) with another mode of operation.

The player will then be able to choose the option that suits him best, or neither of the two options.

**Reply by Branikolog** [reply]
So you're intended to make a slightly different mechanics. Now it's clear.
As I told you somewhere in other discussions, I don't have enough time to create my own options right now. And actually, I think it's a little too early for such global improvements.)
So for now I'm just discussing with you such improvements to find out the conception. Also later people would more deeply understand, while reading this threads, what is the clue of a certain option.
